src/vivarium_nih_us_cvd/components/risk_correlation.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+import pandas as pd
+import scipy
+from vivarium.framework.engine import Builder
+from vivarium.framework.population.manager import SimulantData
+from vivarium_public_health.utilities import EntityString
+
+from vivarium_nih_us_cvd.components.risks import CorrelatedRisk
+from vivarium_nih_us_cvd.constants import paths
+
+
+class RiskCorrelation:
+    """Apply correlation to risk factor propensities"""
+
+    def __repr__(self) -> str:
+        return f"RiskCorrelation"
+
+    ##############
+    # Properties #
+    ##############
+
+    @property
+    def name(self) -> str:
+        return f"risk_correlation"
+
+    #################
+    # Setup methods #
+    #################
+
+    # noinspection PyAttributeOutsideInit
+    def setup(self, builder: Builder) -> None:
+        self.correlation_data = pd.read_csv(paths.FILEPATHS.RISK_CORRELATION)
+        self.risks = [
+            EntityString(risk.name.replace("risk.", ""))
+            for risk in builder.components.get_components_by_type(CorrelatedRisk)
+        ]
+        self.propensity_column_names = [f"{risk.name}_propensity" for risk in self.risks]
+        self.population_view = builder.population.get_view(
+            ["age"] + self.propensity_column_names
+        )
+
+        builder.population.initializes_simulants(
+            self.on_initialize_simulants,
+            creates_columns=self.propensity_column_names,
+            requires_columns=["age"],
+        )
+
+    ##################

```

Comment:
Nit: I'd put the helper functions section below the event-driven methods section.